slabs/migrations.go

```diff
@@ -195,47 +197,21 @@ func sectorsToMigrate(slab Slab, allHosts []hosts.Host, goodContracts []contract
 		// remove contract from the map since we don't want to use it again
 		delete(goodContractMap, *sector.ContractID)
 
-		// NOTE: in testing CIDR checks are disabled because the hosts' network
-		// is an unspecified IPv6 address, in those cases we use the host's
-		// public key to ensure we don't migrate the bad sector to a used host
-		if disableCIDRChecks {
-			usedCIDRs[sector.HostKey.String()] = struct{}{}
-			continue
-		}
-
-		for _, network := range hostsMap[*sector.HostKey].Networks {
-			usedCIDRs[network] = struct{}{}
+		// add the host to the spaced set to ensure we don't use hosts that are
+		// too close to existing hosts
+		if host, ok := hostsMap[*sector.HostKey]; ok {
+			set.Add(host)
 		}
 	}
 
 	// return all hosts with contracts that are good, currently not in use and
-	// don't have overlapping CIDRs, if CIDR checks are disabled we only check
-	// against the host key to ensure unique hosts are used
+	// are sufficiently far apart
 	var candidates []hosts.Host
-	if disableCIDRChecks {
-		for _, contract := range goodContractMap {
-			if _, used := usedCIDRs[contract.HostKey.String()]; used {
-				continue
-			}
-			usedCIDRs[contract.HostKey.String()] = struct{}{}
-			candidates = append(candidates, hostsMap[contract.HostKey])
-		}
-		return toMigrate, candidates
-	}
-
-LOOP:
 	for _, contract := range goodContractMap {
-		host := hostsMap[contract.HostKey]
-		for _, network := range host.Networks {
-			if _, ok := usedCIDRs[network]; ok {
-				continue LOOP
-			}
-		}
-
-		for _, network := range host.Networks {
-			usedCIDRs[network] = struct{}{}
+		if host, ok := hostsMap[contract.HostKey]; ok && set.CanAddHost(host) {
```

Comment:
`Add` returns a boolean right? So this might be slightly nicer

```suggestion
		if host, ok := hostsMap[contract.HostKey]; ok && set.Add(host) {
			candidates = append(candidates, host)
```